docker_mcp/services/stack.py

```
@@ -489,3 +724,102 @@
+            return "0 B"
+        
+        for unit in ['B', 'KB', 'MB', 'GB', 'TB']:
+            if size_bytes < 1024.0:
+                if unit == 'B':
+                    return f"{int(size_bytes)} {unit}"
+                else:
+                    return f"{size_bytes:.1f} {unit}"
+            size_bytes /= 1024.0
+        return f"{size_bytes:.1f} PB"
+    
+    def _extract_expected_mounts(self, compose_content: str, target_appdata: str, stack_name: str) -> list[str]:
```

Ruff (0.12.2)

`_extract_expected_mounts` is too complex (16 > 10)

(C901)